src/backend/cc.d

```diff
@@ -0,0 +1,1674 @@
+    WM_ret_auto     = 15,   //returning address of automatic '%s'
+    WM_ds_ne_dgroup = 16,   //DS is not equal to DGROUP
+    WM_unknown_pragma = 17, //unrecognized pragma
+    WM_implied_ret  = 18,   //implied return at closing '}' does not return value
+    WM_num_args     = 19,   //%d actual arguments expected for %s, had %d
+    WM_before_pch   = 20,   //symbols or macros defined before #include of precompiled header
+    WM_pch_first    = 21,   //precompiled header must be first #include when -H is used
+    WM_pch_config   = 22,
+    WM_divby0       = 23,
+    WM_badnumber    = 24,
+    WM_ccast        = 25,
+    WM_obsolete     = 26,
+
+    // if TARGET_LINUX || TARGET_OSX || TARGET_FREEBSD || TARGET_OPENBSD || TARGET_SOLARIS
+    WM_skip_attribute   = 27, // skip GNUC attribute specification
+    WM_warning_message  = 28, // preprocessor warning message
+    WM_bad_vastart      = 29, // args for builtin va_start bad
+    WM_undefined_inline = 30, // static inline not expanded or defined
+}
+
+// Language for error messages
+enum LANG
+{
+    LANGenglish,
+    LANGgerman,
+    LANGfrench,
+    LANGjapanese,
+}
+
+
+//#if MEMMODELS == 1
+//#define LARGEDATA 0     /* don't want 48 bit pointers */
+//#define LARGECODE 0
+//#endif
+
+//#if SPP || SCPP
+//#include        "msgs2.h"
+//#endif
+//#include        "ty.h"
+//#if TARGET_LINUX || TARGET_OSX || TARGET_FREEBSD || TARGET_OPENBSD || TARGET_SOLARIS
+//#include        "../tk/mem.h"
+//#else
+//#include        "mem.h"
+//#endif
+//#include        "list.h"
+//#include        "vec.h"
+
+//#if SPP
+//#define COMPILER "Preprocessor"
+//#define ACTIVITY "preprocessing..."
+//#elif HTOD
+//#define COMPILER ".h to D Migration Tool"
+//#define ACTIVITY "migrating..."
+//#else
+//#define COMPILER "C/C++ Compiler"
+//#define ACTIVITY "compiling..."
+//#endif
+
+//#ifdef DEBUG
+//#   define debug(a)     (a)
+//#   define debugx(a)    (a)
+//#   define debug_assert assert
+//#else
+//#   define debug(a)
+//#   define debugx(a)
+//#   define debug_assert(e)
+//#endif
+
+/***************************
+ * Print out debugging information.
+ */
+
+//#ifdef DEBUG
+//#define debugmes(s)     (debugw && dbg_printf(s))
+//#define cmes(s)         (debugc && dbg_printf(s))
+//#define cmes2(s,b)      (debugc && dbg_printf((s),(b)))
+//#define cmes3(s,b,c)    (debugc && dbg_printf((s),(b),(c)))
+//#else
+//#define debugmes(s)
+//#define cmes(s)
+//#define cmes2(s,b)
+//#define cmes3(s,b,c)
+//#endif
+
+
+//#define arraysize(array)        (sizeof(array) / sizeof(array[0]))
+
+enum IDMAX = 900;              // identifier max (excluding terminating 0)
+enum IDOHD = 4+1+int.sizeof*3; // max amount of overhead to ID added by
+enum STRMAX = 65000;           // max length of string (determined by
+                               // max ph size)
+
+//enum SC;
+struct Thunk;
+struct token_t;
+//struct param_t;
+//struct block;
+//struct Classsym;
+//struct Nspacesym;
+//struct Outbuffer;
+//struct Aliassym;
+struct dt_t;
```

Comment:
Yes. Unfortunately, those .h files are all interrelated and depend on each other. I can't unwind those problems until they're all in D. Another problem I discovered is the macros were hiding forward references. Those'll have to be dealt with piece by piece.

The first step is this one - get a rote translation to compile and in the build system.